takedown_feats.py

```diff
@@ -0,0 +1,421 @@
+
+        Args:
+            time_str (str): The time string to convert.
+
+        Returns:
+            int: The time in total seconds.
+        """
+
+        if ':' in time_str:
+            minutes, seconds = time_str.split(':')
+            return int(minutes) * 60 + int(seconds)
+        else:
+            return int(time_str)
+        
+    def get_round_time(self, outcome_round, outcome_time, round):
+        """
+        Calculates the total fight time in a specific round.
+
+        Args:
+            outcome_round (int): The round number at the end of the fight.
+            outcome_time (str): The time at which the fight ended in the outcome round.
+            round (str): The round for which the time is to be calculated.
+
+        Returns:
+            float: The total fight time in the specified round in minutes.
+        """
+
+        # Convert outcome_time to seconds
+        outcome_time = self.convert_time_to_seconds(outcome_time)
+        
+        # Handle the "overall" case
+        if round == 0:
+            # Calculate total fight time up to the outcome round
+            total_time = 0
+
+            for r in range(1, outcome_round):
+                total_time += 300  # Add full time for each completed round
+
+            total_time += outcome_time  # Add the time of the last (outcome) round
+
+            return total_time / 60  # Convert to minutes
+
+        # Handle specific rounds
+        if outcome_round == round:
+            return outcome_time/60  # Convert to minutes
+        
+        elif outcome_round > round:
+            return 300/60  # Full round time
+        
+        else:
+            return 0  # Round did not occur
+
+
+    def get_takedowns_return_values(self, stats_by_fight_df, fighter_id, is_percentage, is_defence):
+        """
+        Calculates either the rate or the percentage of takedowns for a fighter.
+
+        Args:
+            stats_by_fight_df (pd.DataFrame): DataFrame containing the statistics of the fights.
+            fighter_id (str): ID of the fighter whose statistics are being calculated.
+            is_percentage (bool): Flag to indicate whether to calculate the percentage (True) or rate (False).
+            is_defence (bool): Flag to indicate whether to calculate for defence (True) or offence (False).
+
+        Returns:
+            float: Calculated rate or percentage of takedowns.
+        """
+
+        # Get the IDs of the fighters in the fights
+        fighter_a_id_vals = stats_by_fight_df.fighter_a_id.values
+        fighter_b_id_vals = stats_by_fight_df.fighter_b_id.values
+
+
+        # Sum up takedowns landed based on whether it's for defence or offence
+        td_landed = stats_by_fight_df[fighter_a_id_vals == fighter_id][f'fighter_a_td_landed' if not is_defence else f'fighter_b_td_landed'].sum() + stats_by_fight_df[fighter_b_id_vals == fighter_id][f'fighter_b_td_landed' if not is_defence else f'fighter_a_td_landed'].sum()
```

Comment:
This line is long, maybe split up between a few lines? In this case it could be beneficial to have if statements to make the code a bit more readable

Reply:
ok i wll split it into two